Makefile

```diff
@@ -82,9 +82,9 @@ lint:
 
 coverage:
 	$(COVERAGE) erase
-	$(COVERAGE) run "--include=$(PACKAGE)/*.py,$(TESTS_DIR)/*.py" --branch -m unittest
-	$(COVERAGE) report "--include=$(PACKAGE)/*.py,$(TESTS_DIR)/*.py"
-	$(COVERAGE) html "--include=$(PACKAGE)/*.py,$(TESTS_DIR)/*.py"
+	$(COVERAGE) run --branch -m unittest
```

Comment:
There is a fundamental difference here that is that all the warnings will be ignored.

Reply:
What about having the ci run the test suite with coverage, 1 stone, 2 birds 😇

Reply:
Or do you think an issue may be revealed with coverage that wasn’t with the regular test suite?

Reply:
> What about having the ci run the test suite with coverage, 1 stone, 2 birds innocent

That mostly works for me. I only looks at coverage from time to time, so I prefer not to add it to the CI for all changes, but am not strongly opposed to it.

Reply:
> What about having the ci run the test suite with coverage, 1 stone, 2 birds innocent

If we do measure coverage on each run, then we should definitely skip the test suite in CI.

Reply:
I agree, I only look at coverage once in a while.

1. When I'm adding new functionality and I want to make darn sure the main idea of the feature is covered.
2. When I'm reviewing someone's code and at a glance, I can see if their functionality is covered -- if not reject it

_Its not required, but its very comfortable when its there_

Reply:
> If we do measure coverage (with warnings turned into errors and -X dev) on each run, then we should definitely skip the test suite in CI.

Naturally. They both do the same thing, just one does.. more?? But there is no need for both to run. (if im understanding you correctly you're saying `test suite` is `make test` and `measure coverage` is `make coverage` right?)

Reply:
The way I understand the status quo is that changes must pass the test suite (without adding warnings), that’s what we verify with CI. People interested can measure the coverage easily (run `make coverage`), but that’s optional.

My expectation is we don’t measure coverage to get faster feedback. In the end, both solutions work well for me.

> test suite is make test and measure coverage is make coverage right?
Yep.

Reply:
The difference between coverage and no coverage seems to be about [~1:10 minutes](https://github.com/FactoryBoy/factory_boy/pull/946#discussion_r908968184), my idea to address your concern is that it can run without coverage to get some of the speed back, and only run coverage in `master` where speed is not an issue. Also if you add "`coverage`" to your commit message it would also trigger it. I'm open to other suggestions

Reply:
One of the major benefits listed in https://github.com/FactoryBoy/factory_boy/pull/952#discussion_r908802134 is the ability to see the coverage on PRs. Running only on master will loose that benefit.

I think we should avoid adding complexity around it. Either we measure coverage or we don’t. More configuration options just become a pain to maintain IMO, it requires more verification (for each configuration set). Adding "coverage" to the commit message to trigger coverage looks like a hidden feature, that can be easily missed by those who would want it, and unintentionally triggered.

Reply:
I agree and I'm nudging towards the "always run coverage" camp, but giving you solutions to your concerns ;)

Reply:
Let’s wait a few days for others to weigh in. I’m not opposed to measuring coverage, my personal preference is against it, mostly because coverage data is not often looked at and slows down the tests pipeline. Having a recipe to measure coverage on a best effort basis (the current situation) is sufficient.
But if @rbarrois or others also find measure coverage on each CI run helpful, I’m happy to follow the group.

Reply:
This thread has been somewhat steered away from the main point of discussion, the reason this started was to add the warning check here. We can merge this now, and continue this conversation over [here](https://github.com/FactoryBoy/factory_boy/pull/946/files#r909369057). And merge this PR, which is still an improvement to what we currently have.